### PR TITLE
Add optional GDScript review hotspot reports

### PR DESCRIPTION
**Pull request by C76GN**
## Summary

Add an explicit `review-hotspots` maintenance command that reports GDScript function locations, branches, loops, control nesting, code lines, and lambda observations. Maintainers can select literal file/directory scopes and inspect stable human-readable or JSON results with per-file source digests.

The report provides no scalar quality score or refactoring threshold and does not enter CI or merge gates. Unsupported syntax and incomplete capture remain visible rather than being presented as complete measurements.

## Contract and Risk

- Change type: feat (repository maintenance tooling).
- Consumer-visible behavior: maintainers can request a bounded current-source report; project runtime/editor behavior is unchanged.
- API or extension contract: additive CLI/report schema version 1; no GDScript API or extension changes.
- Persisted data, package, protocol, or ProjectSettings impact: optional JSON output below `build/`; no project data, package, or settings changes.
- Failure, cancellation, rollback, concurrency, and ownership impact: frozen Git environment and bounded native byte capture; the original deadline covers process-tree quiet confirmation. Cleanup debt is propagated. Pinned file reads reject links, replacement, invalid UTF-8, and budget exhaustion. Per-file digests are not an atomic repository snapshot.
- Compatibility and migration: no migration. The existing `12.0.0-dev.0` development line remains unchanged.
- Observation limits: semicolon statement sequences and multiline lambdas embedded in control headers are explicitly incomplete. Top-N omission is separate from incomplete coverage.

## Verification

- [x] Focused tests cover the changed behavior and failure path: 50 report/parser tests passed, including seven new regression cases for Unicode names, inventory diagnostics/directory markers, and aggregate byte budgets. A real Git nested-repository inventory also completed the requested source scope.
- [x] `python tools/gf_maintenance.py check --suite quick --failed-only`: all 30 checks passed with UTF-8 Python mode.
- [x] `python tools/gf_maintenance.py maintenance-self-test`: 391/391 passed.
- [x] GDScript warning and LSP gates: no `.gd` files changed; existing generated API/knowledge checks passed in quick.
- [x] Independent read-only review of immutable source snapshots completed; identified parser and process-ownership issues were fixed and regression-tested.
- [x] Real CLI observations: 57 math scripts / 1,097 functions completed; the full source inventory covered 972 scripts / 21,819 functions and explicitly reported the two unsupported control-header lambda files.
- [x] Draft PR feedback completed through `GF draft gate`.
- [x] Ready PR CI completed through `GF repository policy` and `GF merge gate` for c75604e4, including full validation and Windows process supervision.

## Documentation and Release

- [x] The maintainer guide documents usage, metric definitions, scope, budgets, status, and limitations. This is internal maintenance functionality, so no public product changelog entry is needed.
- [x] Development version impact is correct.
- [x] This PR does not create a tag or publish a release.

### CODE REVIEW

File: tools/gf_review_hotspots.py [hunk 1]
```diff
@@ -0,0 +1,502 @@
...
+METRIC_DEFINITIONS = {
+	"branch_count": "Explicit if/elif/else headers plus one per match arm, including wildcard; no boolean-operator or guard bonus.",
+	"loop_count": "Explicit for/while headers; comprehensions and call semantics are not inferred.",
+	"match_arm_count": "One per match arm, regardless of alternative patterns or guards.",
+	"max_nesting": "Maximum enclosing control-block depth including match and arms; excludes function/class/lambda/container scopes.",
+	"effective_code_lines": "Distinct body physical lines containing masked structural code; excludes signature, comments, blank and literal-only lines.",
+	"lambda_count": "Lambda declarations within the named function, including nested lambda bodies; no separate lambda rows.",
+	"conditional_expression_count": "Non-header if tokens, observed separately from statement branches.",
+}
+_TOKEN = re.compile(r"\w+|->|:=|[^\s]", re.UNICODE)
```
[the remark this answers is not in the document]
Comment (C76GN) [reply tools/gf_review_hotspots.py:76]: Fixed in c75604e4. Tokenization now joins adjacent Unicode identifier continuation fragments in linear time and preserves the full raw declaration name even when the shared declaration parser returns a prefix. Added decomposed-name, inner-class/static-method, and long-combining-mark budget regressions; all 31 parser tests pass.

File: tools/gf_maintenance.py [hunk 2]
```diff
@@ -2352,6 +2372,52 @@ def write_internal_complete_output_evidence_report(
...
+			result.return_code != 0 or result.stderr
+			or result.stdout_truncated or result.stderr_truncated
+			or result.timed_out or result.output_drain_failed or not result.cleanup_complete
+		):
+			raise ValueError("review_hotspots.git_inventory_failed")
+		inventory = gf_review_hotspot_report.parse_inventory(result.stdout)
+	except (OSError, ValueError, subprocess.TimeoutExpired) as error:
+		if exception_has_cleanup_debt(error):
+			raise
+		report["issues"].append({"code": "review_hotspots.git_inventory_failed"})
```
[the remark this answers is not in the document]
Comment (C76GN) [reply tools/gf_maintenance.py:2415]: Fixed in c75604e4. Inventory parsing now has a separate error boundary, preserving UTF-8, path, and truncated-inventory codes while subprocess failures retain git_inventory_failed. The existing process deadline and cleanup-debt handling are unchanged; dedicated CLI regressions pass.

File: tools/gf_review_hotspot_report.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+		raise ValueError("review_hotspots.incomplete_inventory")
+	try:
+		paths = payload.decode("utf-8", errors="strict").split("\0")[:-1]
+	except UnicodeDecodeError:
+		raise ValueError("review_hotspots.invalid_inventory_utf8") from None
+	if len(paths) > MAX_INVENTORY_PATHS:
+		raise ValueError("review_hotspots.inventory_limit")
+	for path in paths:
+		if "\\" in path or normalize_scope(path) != path:
+			raise ValueError("review_hotspots.invalid_inventory_path")
```
[the remark this answers is not in the document]
Comment (C76GN) [reply tools/gf_review_hotspot_report.py]: Fixed in c75604e4. Canonical trailing-slash directory markers are validated and excluded without traversing the nested repository; malformed and escaping markers still fail. Verified both regression fixtures and a real git ls-files inventory containing nested/ followed by src/a.gd, with the requested src report completing successfully.

File: tools/gf_review_hotspot_report.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+	total_bytes = 0
+	functions: list[dict[str, Any]] = []
+	for path in sorted(selected):
+		if time.monotonic() >= deadline:
+			report["issues"].append({"code": "review_hotspots.deadline"})
+			break
+		try:
+			payload = gf_path_security.read_pinned_regular_file(
+				root, path, max_bytes=min(MAX_FILE_BYTES, MAX_TOTAL_BYTES - total_bytes),
+			)
```
[the remark this answers is not in the document]
Comment (C76GN) [reply tools/gf_review_hotspot_report.py:149]: Fixed in c75604e4. When the remaining aggregate allowance is the limiting capture budget, exhaustion reports review_hotspots.total_bytes_limit and stops scanning. Reads remain bounded by the smaller allowance; exact exhaustion performs no further read, and the independent per-file limit keeps its original diagnostic. Regression tests assert both codes and actual read caps/counts.
